src/main/java/org/apache/maven/plugins/jlink/JLinkMojo.java

```diff
@@ -464,12 +464,16 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         }
     }
 
-    private List<File> getCompileClasspathElements(MavenProject project) {
+    List<File> getCompileClasspathElements(MavenProject project) {
         List<File> list = new ArrayList<>(project.getArtifacts().size() + 1);
 
         for (Artifact a : project.getArtifacts()) {
-            getLog().debug("Artifact: " + a.getGroupId() + ":" + a.getArtifactId() + ":" + a.getVersion());
-            list.add(a.getFile());
+            boolean skip = "pom".equals(a.getType());
+            getLog().debug("Adding artifact: " + a.getGroupId() + ":" + a.getArtifactId() + ":" + a.getVersion()
+                    + (skip ? " (skipping)" : ""));
+            if (!skip) {
+                list.add(a.getFile());
+            }
```

Comment:
Thinking out loudly, maybe in the future more deps will be skipped. Maybe (just maybe) refactor out a method here which currently only contains `"pom".equals(a.getType())`, e.g. `shouldSkip(artifact)`. Then rename `skip` to `shouldSkip`.

Makes it future proof, and the renaming of the variable will tell maintainers why it should be skipped (at least a little).